WordPress/openverse
# Fix the full-height layout

## Fixes



Fixes #1159 by @obulat
Fixes #504 by @sarayourfriend 
Fixes #2062 by @AetherUnbound 
Fixes #594 by @obulat

## Description


This PR adds a new layout file for search pages.

This layout is different from content pages because needs the sidebar to be fixed, as well as the header. We were using `fixed` for its position, but it required adding a top margin to account for the sticky header.

This implementation uses the following CSS grid:
```
+--------------+---------+
|            header      |
+--------------+---------+
| main content | sidebar |
| footer       |         |
+--------------+---------+
```
This means that we no longer have to make the header or the sidebar sticky, and they are placed correctly vertically and not covered with other elements (see the changes in `search-with-banners/page-with-all-banners` snapshots for how it fixes the issue).

This also fixes the cut-off of the modal's bottom part and the scroll to top button on mobile screens.

This layout changes the scrollable element from the document to "main-content" (the part where the results are displayed, not the header or sidebar). This means that Playwright cannot create the full-page snapshots using `fullPage` prop, and this PR adds a utility that increases the viewport for no results page so that all of its content fits into the page to balance it.

## Testing Instructions

Run the app and check that the search page works correctly both on desktop and mobile screens.
On desktop:
- Try opening all banners by going to localhost:8443/af/search/?q=cat&referrer=creativecommons.org and opening the filters sidebar: its headings should be visible (unlike on https://openverse.org/af/search/?q=cat&referrer=creativecommons.org)
- Try searching - the recent searches should be visible above the page content
- If you scroll down (both with filters open and closed), the scroll to top button should be visible

On mobile:
- The content settings mobile should be fully visible, and the buttons should not be cut off.
- If you scroll down, the scroll to top button should be visible and not cut off.

The content layout (the pages like `/about` and single result pages) should also work correctly. They did not have a scroll to top button - do we want it there?

When you navigate from the single result view to the search results on desktop, there should be no flickering described in #504.


## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
